Makefile.doc

```diff
@@ -82,7 +82,7 @@ doc: sphinx refman tutorial rectutorial stdlib $(INDEXURLS)
 
 sphinx: coq
 	$(SHOW)'SPHINXBUILD doc/sphinx'
-	$(HIDE)COQBIN="$(PWD)/bin" $(SPHINXBUILD) -b html $(ALLSPHINXOPTS) doc/sphinx $(SPHINXBUILDDIR)/html
+	$(HIDE)COQBIN="$(PWD)/bin" $(SPHINXBUILD) -W -b html $(ALLSPHINXOPTS) doc/sphinx $(SPHINXBUILDDIR)/html
```

Comment:
This should have been linked with the `-warn-error` option of `configure` IMHO.

Reply:
I thought about it, but decided against it. Sphinx "warnings" are really things that should be errors (and would be in OCaml), so the analogy does really work.

Reply:
Well, I'm not so convinced that it is still true with the new warnings added with #7374.